clang/lib/CConv/DeclRewriter.cpp

```diff
@@ -52,6 +52,23 @@ void DeclRewriter::rewriteDecls(ASTContext &Context, ProgramInfo &Info,
   for (const auto &D : Context.getTranslationUnitDecl()->decls()) {
     TRV->TraverseDecl(D);
     SVI.TraverseDecl(D);
+    if (const auto &TD  = dyn_cast<TypedefDecl>(D)) {
+      auto PSL = PersistentSourceLoc::mkPSL(TD, Context);
+      if (!TD->getUnderlyingType()->isBuiltinType()) { // Don't rewrite base types like int
```

Comment:
I would expect that this code will basically match the code you would use for rewriting any type. Can you recursively invoke that code? In particular, if I had
```
int **x = ...
int y = ...
```
then rewriting would look at each variable's type and figure out how to rewrite it. It might rewrite the first because it's a pointer and is checked, but will not rewrite the second, because it's not a pointer. I would think that all of the myriad reasoning we use for normal declarations would just apply the RHS of a typedef, no? If so, it seems to me that you need to recursively invoke the code somehow to do that, so you are sure you hit every case. Am I making sense?